File: BussinesLogic/Servicios/ServicioLineaNegocio.cs
using DataAccess.StoredProcedures;
using Dominio.Dto;
using Dominio.Entiti;
using Dominio.Interfaces.Aplicacion.LineaNegocio;
using Dominio.Interfaces.Infraestructura.BaseDatos;
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;

namespace BussinesLogic.Servicios
{
    public class ServicioLineaNegocio : IServicioLineaNegocio
    {
        private readonly IRepositorioLineaNegocio repositorioLineaNegocio;
        public ServicioLineaNegocio(IRepositorioLineaNegocio repositorioLineasNegocios)
        {
            this.repositorioLineaNegocio = repositorioLineasNegocios;
        }


        public async Task<DtoRespuestaSP> InsertarLineaNegocio(EntitiLineaNegocio entitiLineaNegocio)
        {
            return await this.repositorioLineaNegocio.InsertarLineaNegocio(entitiLineaNegocio);
        }
        public async Task<DtoRespuestaSP> ActualizarLineaNegocio(EntitiLineaNegocio entitiLineaNegocio)
        {
            return await this.repositorioLineaNegocio.ActualizarLineaNegocio(entitiLineaNegocio);
        }
        public async Task<DtoRespuestaSP> EliminarLineaNegocio(int idLinea)
        {
            return await this.repositorioLineaNegocio.EliminarLineaNegocio(idLinea);
        }
        public async Task<DtoLineaNegocio> ObtenerLineaNegocioPorID(int idLinea)
        {
            return await this.repositorioLineaNegocio.ObtenerLineaNegocioPorID(idLinea);
        }

        public async Task<List<DtoLineaNegocio>> ObtenerLineaNegocio()
        {
            return await this.repositorioLineaNegocio.ObtenerLineaNegocio();
        }


    }
}

File: CX/ClientApp/src/servicios/ServicioLineaNegocio.js
import { ProcesarDatosApi } from "./ApiFetch";
const controlador = "lineanegocio";

export const AgregarLineaNegocio = async (data) => {
    const url = `${controlador}/insertarlineanegocio`;
    return await ProcesarDatosApi('POST', url, data);
}

export const ActualizarLineaNegocio = async (data) => {
    const url = `${controlador}/actualizarlineanegocio`;
    return await ProcesarDatosApi('PUT', url, data);
}

export const InactivarLineaNegocio = async (id) => {
    const url = `${controlador}/eliminarlineanegocio?idLinea=${id}`;
    return await ProcesarDatosApi('DELETE', url);
}

export const ObtenerLineaNegocioPorId = async (id) => {
    const url = `${controlador}/obtenerlineanegocioporid/${id}`;
    return await ProcesarDatosApi('GET', url);
}

export const ObtenerLineaNegocio= async () => {
    const url = `${controlador}/ObtenerLineaNegocio`;
    return await ProcesarDatosApi('GET', url);
}
